src/microdot.py

```diff
@@ -212,6 +212,8 @@ def __init__(self, app, client_addr, method, url, http_version, headers,
             if header == 'Content-Length':
                 self.content_length = int(value)
             elif header == 'Content-Type':
+                if value:
+                    value = value.split(';')[0]
```

Comment:
This is not a good solution. Basically you are dropping the encoding altogether, making it impossible to use it later on.